modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2DeviceFlowTestCase.java

```diff
@@ -0,0 +1,338 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.COMMON_AUTH_URL;
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.USER_AGENT;
+
+public class OAuth2DeviceFlowTestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private AuthenticatorClient logManger;
+    private String adminUsername;
+    private String adminPassword;
+    private String accessToken;
+    private String sessionDataKeyConsent;
+    private String sessionDataKey;
+    private String authorizationCode;
+    private String consumerKey;
+    private String consumerSecret;
+    private String userCode;
+    private String deviceCode;
+
+
+    private static final String PLAYGROUND_RESET_PAGE = "http://localhost:" + CommonConstants.DEFAULT_TOMCAT_PORT +
+            "/playground2/oauth2.jsp?reset=true";
+    private DefaultHttpClient client;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.SUPER_TENANT_USER);
+        logManger = new AuthenticatorClient(backendURL);
+        adminUsername = userInfo.getUserName();
+        adminPassword = userInfo.getPassword();
+        logManger.login(isServer.getSuperTenant().getTenantAdmin().getUserName(),
+                isServer.getSuperTenant().getTenantAdmin().getPassword(),
+                isServer.getInstance().getHosts().get("default"));
+        client = new DefaultHttpClient();
+
+        setSystemproperties();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        deleteApplication();
+        removeOAuthApplicationData();
+
+    }
+
+    @Test(groups = "wso2.is", description = "Check Oauth2 application registration")
+    public void testRegisterApplication() throws Exception {
+
+        OAuthConsumerAppDTO appDto = createApp();
+        Assert.assertNotNull(appDto, "Application creation failed.");
+
+        consumerKey = appDto.getOauthConsumerKey();
+        Assert.assertNotNull(consumerKey, "Application creation failed.");
+
+        consumerSecret = appDto.getOauthConsumerSecret();
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request without redirect_uri param", dependsOnMethods
+            = "testRegisterApplication")
+    public void testSendDeviceAuthorize() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("client_id", consumerKey));
+        AutomationContext automationContext = new AutomationContext("IDENTITY", TestUserMode.SUPER_TENANT_ADMIN);
+        String deviceAuthEndpoint = automationContext.getContextUrls().getBackEndUrl()
+                .replace("services/", "oauth2/device_authorize");
+        JSONObject responseObject = responseObjectNew(urlParameters,deviceAuthEndpoint);
+        deviceCode = responseObject.get("device_code").toString();
```

Comment:
`device_code`: repeating terms like this, can be declared as constants and reused.